### ISSUE TITLE
Test failures on s390x architecture with 1.10.0

### ISSUE THREAD

**Issue by swt2c** [pocoproject/poco#2909]
I'm seeing a few test failures on s390x with 1.10.0.  I suspect these are possible endianness issues as s390x is big endian?

Net:
```
6: !!!FAILURES!!!
6: Runs: 253   Failures: 1   Errors: 0
6: 
6: There was 1 failure: 
6:  1: N7CppUnit10TestCallerI19NTLMCredentialsTestEE.testParseChallengeMessageWithTargetInfo
6:     "msg2.target == "DOMAIN""
6:     in "/builddir/build/BUILD/poco-poco-1.10.0-release/Net/testsuite/src/NTLMCredentialsTest.cpp", line 204
6: 
11/12 Test  #6: Net ..............................***Failed   75.68 sec
```
Foundation:
```
1: !!!FAILURES!!!
1: Runs: 786   Failures: 4   Errors: 0
1: 
1: There were 4 failures: 
1:  1: N7CppUnit10TestCallerI14SHA2EngineTestEE.testSHA224
1:     "DigestEngine::digestToHex(engine.digest()) == "23097d223405d8228642a477bda255b32aadbce4bda0b3f7e36c9da7""
1:     in "/builddir/build/BUILD/poco-poco-1.10.0-release/Foundation/testsuite/src/SHA2EngineTest.cpp", line 40
1:  2: N7CppUnit10TestCallerI14SHA2EngineTestEE.testSHA256
1:     "DigestEngine::digestToHex(engine.digest()) == "ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad""
1:     in "/builddir/build/BUILD/poco-poco-1.10.0-release/Foundation/testsuite/src/SHA2EngineTest.cpp", line 59
1:  3: N7CppUnit10TestCallerI14SHA2EngineTestEE.testSHA384
1:     "DigestEngine::digestToHex(engine.digest()) == "cb00753f45a35e8bb5a03d699ac65007272c32ab0eded1631a8b605a43ff5bed8086072ba1e7cc2358baeca134c825a7""
1:     in "/builddir/build/BUILD/poco-poco-1.10.0-release/Foundation/testsuite/src/SHA2EngineTest.cpp", line 78
1:  4: N7CppUnit10TestCallerI14SHA2EngineTestEE.testSHA512
1:     "DigestEngine::digestToHex(engine.digest()) == "ddaf35a193617abacc417349ae20413112e6fa4e89a97ea20a9eeee64b55d39a2192992a274fc1a836ba3c23a3feebbd454d4423643ce80e2a9ac94fa54ca49f""
1:     in "/builddir/build/BUILD/poco-poco-1.10.0-release/Foundation/testsuite/src/SHA2EngineTest.cpp", line 97
1: 
12/12 Test  #1: Foundation .......................***Failed  112.35 sec
```

**Comment by swt2c** [comment]
FYI, it appears that Travis now supports s390x as an architecture:
https://docs.travis-ci.com/user/multi-cpu-architectures

**Comment by obiltschnig** [comment]
fixed, see #2912, #2911
